lib/setup.c

```diff
@@ -921,6 +926,37 @@ ___SIZE_TS requested_words_still;)
 }
 
 
+___EXP_FUNC(___BOOL, ___fdset_resize)
+   ___P((___processor_state ___ps,
+         int fd),
+        (___ps,
+         fd)
+___processor_state ___ps;
+int fd;)
+{
+#ifdef USE_poll
```

Comment:
i am also adding a check for overflow in the case of select.
We can't resize the fd_set, but we can at least avoid crashes when the file descriptor is used.